box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java

```diff
@@ -530,18 +541,62 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
                 public void run() {
                     //TODO decide whether to use two views for crossfading animation.
                     imageView.setImageBitmap(bitmap);
+                    ImageLoadListener listener = ViewData.getImageLoadListener(imageView);
+                    if(listener != null) {
+                        listener.onSuccess();
+                    }
                 }
             });
         }
 
     }
 
-
-
     private boolean isRequestStillApplicable(final BoxRequest request, final ImageView view) {
         return view.getDrawable() instanceof LoaderDrawable &&
                ((LoaderDrawable) view.getDrawable()).matchesRequest(request);
 
     }
 
+    /**
+     * Interface to listen for Image load, so that the calling component can
+     *  act upon error or success
+     */
+    public interface ImageLoadListener {
+        void onSuccess();
+        void onError();
+    }
+
+    /**
+     * Internal class to keep ImageView data information, like type of thumbnail to show, and image load listener
+     */
+    private static class ViewData {
+        private ImageLoadListener mListener;
+        private String mImageType;
+
+        ViewData(String type, ImageLoadListener listener) {
+            mImageType = type;
+            mListener = listener;
+        }
+
+        static ImageLoadListener getImageLoadListener(ImageView view) {
+            if(view != null) {
+                ViewData data = (ViewData) view.getTag();
+                if(data != null) {
+                    return data.mListener;
+                }
+            }
+            return null;
```

Comment:
Should we use the null object pattern so the caller never have to worry if this function return null?

Reply:
I like that.... let me add this. Thanks!